### DISCUSSION TITLE
Trame Examples with animations

### DISCUSSION THREAD

**Discussion by munirakhtar** [Kitware/trame#20] [Trame/VTK] [answered]
Are there any online trame examples including animations with different scalar values (e.g. Temperature, Pressure) for each cell (elements) at different time steps ?

**Comment by jourdain** [comment]
An example is available [here](https://github.com/Kitware/trame/tree/master/examples/ParaView/TimeAnimation). With the car crash files, most the of performance limitation is due to IO which can be greatly improved with custom file handling/loading.

**Comment by munirakhtar** [comment]
Dear Sébastien, Thanks alot:there is a module in app.py of TimeAnimation example in tramefrom paraview.web import venvand there is a comment about it #Available in PV 5.10-RC2+. I could only see Paraview 5.10 RC1 on download site https://www.paraview.org/download/ so where can we get Paraview 5.10 RC2 or above----‐------------------Munir AkhtarSOCO engineers GmbHOn 11/24/21, 03:28 Sebastien Jourdain ***@***.***> wrote:

   
  An example is available here. With the car crash files, most the of performance limitation is due to IO which can be greatly improved with custom file handling/loading.

**Reply by jourdain** [reply]
That [README](https://github.com/Kitware/trame/tree/master/examples/ParaView) explain how to use a local version of that file before RC2.

**Comment by mirabdullahyaser** [comment]
Hi @jourdain, I am working with Munir on the Time Animation example of Trame. I have tested the new time animation example that is working fine however when we change the color attribute in the car crash example then there is no change in the color of mesh and we get the error in the code. I have also attached the image of the error. Thanks!
![Trame Animation error](https://user-images.githubusercontent.com/42169766/143452183-c58a166d-b908-4150-ad4d-a25a75f342c8.png)

**Comment by jourdain** [comment]
Can you describe what you've done to run it with your data?

**Comment by jourdain** [comment]
Just tried again with your data and switching fields is just working fine for me
![car-1](https://user-images.githubusercontent.com/294518/143499156-cd832e97-66d1-43f3-add7-f25720dab014.jpg)
![car-2](https://user-images.githubusercontent.com/294518/143499158-e040a8a3-b03c-449b-b637-2bcbb96d31ba.jpg)
.

**Comment by mirabdullahyaser** [comment]
I did the following steps:
1. removed the first line of code i.e. from paraview.web import venv
2. placed venv.py file from the trame repo to the current folder and added import venv to app.py to use this venv file
3. placed vtu files in the current folder having app.py and car-crash.json
4. made .pvenv according to your given steps and installed trame in it
5. run app.py with the following command
/ParaView-5.10.0-RC1-MPI-Linux-Python3.9-x86_64/bin/pvpython ./TimeAnimation/app.py --venv .pvenv --data ./TimeAnimation/car-crash.json

I have noticed that in the callback function update_color_by() variable active_array was getting the name of parameters like "Max. Pl. Strain" or "Von Mises Stress" or "Element id" and you were passing it as the index of array fields that should actually get the index number as an integer (0, 1, 2). So, I changed the function according to the variables in my data files to the following and it is working fine now.

![Screenshot from 2021-11-26 10-55-12](https://user-images.githubusercontent.com/42169766/143533895-eff7e456-23f7-40f4-a1a1-948e25fa13a0.png)

**Comment by jourdain** [comment, marked as answer]
Good catch... I was not using the one in the repo. I just pushed a fix to match what the code is expecting.
```
+++ b/examples/ParaView/TimeAnimation/car-crash.json
@@ -16,8 +16,8 @@
         "CellArrayStatus": ["Max. Pl. Strain", "Von Mises Stress", "Element id"]
     },    
     "fields": [
-        { "text": "Max. Pl. Strain", "value":"Max. Pl. Strain", "location": "CELLS", "range": [0.0001, 0.05] },
-        { "text": "Von Mises Stress", "value":"Von Mises Stress", "location": "CELLS" , "range": [50, 500]},
-        { "text": "Element id", "value":"Element id", "location": "CELLS" , "range": [2000000, 4000000] }
+        { "text": "Max. Pl. Strain", "value": 0, "location": "CELLS", "range": [0.0001, 0.05] },
+        { "text": "Von Mises Stress", "value": 1, "location": "CELLS" , "range": [50, 500]},
+        { "text": "Element id", "value": 2, "location": "CELLS" , "range": [2000000, 4000000] }
     ]
 }
```
